### DISCUSSION TITLE
Time Series Cherenkov ADC Data Improved Analysis

### DISCUSSION THREAD

**Discussion by engineer9001** [williamgilpin/cphy#17] [projects]
My Final project proposal is to improve NOvA Test Beam Cherenkov reconstruction. Currently, a Constant Fraction Discriminator is used to determine signal start, peak, and end times, but it is relatively often that these reconstructed values are visually obviously off from what they should be and I'd like to improve this.

For now, I believe that I will be recreating a CFD that can hopefully be less subject to issues and then I'd like to look more into this data and see if a dynamic time-warp analysis could possibly be used to gain more information from these signals than we already do.

**Comment by williamgilpin** [comment]
https://scikit-hep.org/
+ Lots of utilities for working with ROOT data

https://tslearn.readthedocs.io/en/stable/user_guide/dtw.html
+ Working with time series, DTW implementation

https://scikit-learn.org/stable/modules/generated/sklearn.manifold.MDS.html
+ Can pass this embedding method a "precomputed" distance matrix, such as one from DTW
